### DISCUSSION TITLE
syslog-ng service not starting up

### DISCUSSION THREAD

**Discussion by bishnujena** [syslog-ng/syslog-ng#3977] [Q&A] [answered]
Hi all,
below is the parameters adding in the "/etc/syslog-ng/conf.d/basic.conf" , while restarting the service it is not starting up. Please help.

Platform
NAME="Red Hat Enterprise Linux Server"
VERSION="7.9 (Maipo)"

>> Below is the suggested configuration in syslog-ng website, I have configured in "/etc/syslog-ng/conf.d/basic.conf" file.

# source
source s_linux {
        network(
                ip("[xx.xx.xx.xx](callto:xx.xx.xx.xx)")
                transport("tcp")
                port(51401)
                );
};

# Destination
destination d_linux_local {
        file("/var/log/message_${HOST}");
};

# log
log {
        source(s_linux);
        destination(d_linux_local);
};

****** While I am commenting  "source(s_linux);" service is starting successfully.**** 

# service syslog-ng restart
Redirecting to /bin/systemctl restart syslog-ng.service
Job for syslog-ng.service failed because the control process exited with error code. See "systemctl status syslog-ng.service" and "journalctl -xe" for details.

# service syslog-ng status
Redirecting to /bin/systemctl status syslog-ng.service
● syslog-ng.service - System Logger Daemon
   Loaded: loaded (/usr/lib/systemd/system/syslog-ng.service; enabled; vendor preset: enabled)
   Active: failed (Result: start-limit) since Wed 2022-04-06 12:38:28 IST; 17s ago
     Docs: man:syslog-ng(8)
  Process: 2468 ExecStart=/usr/sbin/syslog-ng -F -p /var/run/syslogd.pid (code=exited, status=2)
 Main PID: 2468 (code=exited, status=2)
   Status: "Starting up... (Wed Apr  6 12:38:28 2022"

Apr 06 12:38:28 localhost.localdomain systemd[1]: syslog-ng.service failed.
Apr 06 12:38:28 localhost.localdomain systemd[1]: syslog-ng.service holdoff time over, scheduling restart.
Apr 06 12:38:28 localhost.localdomain systemd[1]: Stopped System Logger Daemon.
Apr 06 12:38:28 localhost.localdomain systemd[1]: start request repeated too quickly for syslog-ng.service
Apr 06 12:38:28 localhost.localdomain systemd[1]: Failed to start System Logger Daemon.
Apr 06 12:38:28 localhost.localdomain systemd[1]: Unit syslog-ng.service entered failed state.
Apr 06 12:38:28 localhost.localdomain systemd[1]: syslog-ng.service failed.
Apr 06 12:38:28 localhost.localdomain systemd[1]: start request repeated too quickly for syslog-ng.service
Apr 06 12:38:28 localhost.localdomain systemd[1]: Failed to start System Logger Daemon.
Apr 06 12:38:28 localhost.localdomain systemd[1]: syslog-ng.service failed.

# sestatus
SELinux status:                 disabled

# syslog-ng -V
syslog-ng 3.5.6
Installer-Version: 3.5.6
Revision:
Compile-Date: Dec 30 2015 19:57:24
Available-Modules: affile,afprog,afsocket-notls,afsocket-tls,afsocket,afstomp,afuser,basicfuncs,confgen,cryptofuncs,csvparser,dbparser,linux-kmsg-format,syslogformat,system-source
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
Enable-Pcre: on

# journalctl -u syslog-ng
Starting System Logger Daemon...
Apr 06 12:26:27 localhost.localdomain systemd[1]: syslog-ng.service: main process exited, code=exited, status=2/INVALIDARGUMENT
Apr 06 12:26:27 localhost.localdomain systemd[1]: Failed to start System Logger Daemon.
Apr 06 12:26:27 localhost.localdomain systemd[1]: Unit syslog-ng.service entered failed state.
lines 1-47

**Comment by faxm0dem** [comment, marked as answer]
Try starting syslog-ng in the foreground, you'll see more information on the reason why it's failing:
```
syslog-ng -Fdve
```

**Reply by bishnujena** [reply]
Thank you very much, I got the error and service restarted. I had missed a ; in the configuration file.

**Reply by bishnujena** [reply]
Also I have changed the IP address as 0.0.0.0 in the below section then syslog-ng service started. THANK YOU !!!

# source
source s_linux {
network(
ip("xx.xx.xx.xx")
transport("tcp")
port(51401)
);
};
